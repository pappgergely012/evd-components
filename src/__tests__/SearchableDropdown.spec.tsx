/**
 * @jest-environment jsdom
 */

import React from "react";
import { SearchableDropdown } from '../index';
import renderer, { act } from "react-test-renderer";

const cities = [
  { city: 'New York', id: '1' },
  { city: 'Boston', id: '2' }
]

const component = renderer.create(
  <SearchableDropdown
    data={cities}
    placeholder="Cities"
    labelExtractor={(item): string => item.city}
    keyExtractor={(item): string => item.id}
    onChange={(selected) => console.log('value come from searchabledropdown change is ' + selected)}
    activeColor="#333"
  />
);

test("Dropdown open test", () => {
  const instance = component.root;

  expect(instance.props.placeholder).toBe("Cities");
  expect(instance.props.data).toBeInstanceOf(Array);

  act(() => {
    const inst: any = component.toJSON();
    const input: any = inst.children[0].children[0].children[1];

    input.props.onFocus();
  });

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test("Dropdown item clicked test", () => {
  const instance = component.root;

  expect(instance.props.placeholder).toBe("Cities");
  expect(instance.props.data).toBeInstanceOf(Array);

  act(() => {
    const inst: any = component.toJSON();
    const input: any = inst.children[0].children[0].children[1];
    const listItem: any = inst.children[1].children[0];

    input.props.onFocus();
    listItem.props.onClick();
  });

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
