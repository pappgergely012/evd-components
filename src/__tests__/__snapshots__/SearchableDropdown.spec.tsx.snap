// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown item clicked test 1`] = `
<div
  className="dropdown dropdown--searchable full"
>
  <div
    className="full-input-outer"
  >
    <div
      className="input-container focused not-empty"
    >
      <label
        className="input-placeholder"
        style={
          Object {
            "color": "#333",
          }
        }
      >
        Cities
      </label>
      <input
        className="input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "borderColor": "#333",
            "boxShadow": "0 4px 4px rgba(#333, .4)",
          }
        }
        value="New York"
      />
    </div>
  </div>
</div>
`;

exports[`Dropdown open test 1`] = `
<div
  className="dropdown dropdown--searchable full"
>
  <div
    className="full-input-outer"
  >
    <div
      className="input-container focused border-bottom-off "
    >
      <label
        className="input-placeholder"
        style={
          Object {
            "color": "#333",
          }
        }
      >
        Cities
      </label>
      <input
        className="input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "borderColor": "#333",
            "boxShadow": "0 4px 4px rgba(#333, .4)",
          }
        }
        value=""
      />
    </div>
  </div>
  <div
    className="dropdown-list"
    style={
      Object {
        "border": "1px solid #333",
        "borderTop": "0px",
      }
    }
  >
    <div
      className="dropdown-item"
      onClick={[Function]}
    >
      New York
    </div>
    <div
      className="dropdown-item"
      onClick={[Function]}
    >
      Boston
    </div>
  </div>
</div>
`;
